src/server/pachyderm_test.go

```diff
@@ -10968,19 +10965,276 @@ time.Sleep(5)
+		require.NoError(t, c.Fsck(false, func(resp *pfs.FsckResponse) error {
+			if resp.Error != "" {
+				return errors.New(resp.Error)
+			}
+			return nil
+		}))
+	})
+	t.Run("SpoutAuthEnabledAfter", func(t *testing.T) {
+		tu.DeleteAll(t)
+		c := tu.GetPachClient(t)
+
+		dataRepo := tu.UniqueString("TestSpoutAuthEnabledAfter_data")
+		require.NoError(t, c.CreateRepo(dataRepo))
+
+		// create a spout pipeline
+		pipeline := tu.UniqueString("pipelinespoutauthenabledafter")
+		_, err := c.PpsAPIClient.CreatePipeline(
+			c.Ctx(),
+			&pps.CreatePipelineRequest{
+				Pipeline: client.NewPipeline(pipeline),
+				Transform: &pps.Transform{
+					Cmd: []string{"/bin/sh"},
+					Stdin: []string{
+						"while [ : ]",
+						"do",
+						"sleep 0.1",
+						"pachctl auth whoami &> whoami",
+						basicPutFile("./whoami*"),
+						"done"},
+				},
+				Spout: &pps.Spout{}, // this needs to be non-nil to make it a spout
+			})
+		require.NoError(t, err)
+
+		// get 5 succesive commits
+		iter, err := c.SubscribeCommit(pipeline, "master", nil, "", pfs.CommitState_FINISHED)
+		require.NoError(t, err)
+
+		for i := 0; i < 5; i++ {
+			commitInfo, err := iter.Next()
+			require.NoError(t, err)
+			files, err := c.ListFile(pipeline, commitInfo.Commit.ID, "")
+			require.NoError(t, err)
+			require.Equal(t, 1, len(files))
+		}
+
+		// now let's authenticate, and make sure the spout fails due to a lack of authorization
+		tu.ClearPachClientState(t)
+		c = tu.GetAuthenticatedPachClient(t, tu.AdminUser)
+		defer tu.DeleteAll(t)
+
+		// make sure we can delete commits
+		err = c.DeleteCommit(pipeline, "master")
+		require.NoError(t, err)
+
+		// now let's update the pipeline and make sure it works again
+		_, err = c.PpsAPIClient.CreatePipeline(
+			c.Ctx(),
+			&pps.CreatePipelineRequest{
+				Pipeline: client.NewPipeline(pipeline),
+				Transform: &pps.Transform{
+					Cmd: []string{"/bin/sh"},
+					Stdin: []string{
+						"while [ : ]",
+						"do",
+						"sleep 2",
+						"date > date",
+						basicPutFile("./date*"),
+						"done"},
+				},
+				Update:    true,
+				Reprocess: true,         // to ensure subscribe commit will only read commits since the update
+				Spout:     &pps.Spout{}, // this needs to be non-nil to make it a spout
+			})
+		require.NoError(t, err)
+
+		// get 5 succesive commits
+		iter, err = c.SubscribeCommit(pipeline, "master", nil, "", pfs.CommitState_FINISHED)
+		require.NoError(t, err)
+
+		for i := 0; i < 5; i++ {
+			commitInfo, err := iter.Next()
+			require.NoError(t, err)
+			files, err := c.ListFile(pipeline, commitInfo.Commit.ID, "")
+			require.NoError(t, err)
+			require.Equal(t, 1, len(files))
+		}
+
+		// finally, let's make sure that the provenance is in a consistent state after running the spout test
+		require.NoError(t, c.Fsck(false, func(resp *pfs.FsckResponse) error {
+			if resp.Error != "" {
+				return errors.New(resp.Error)
+			}
+			return nil
+		}))
+	})
+	t.Run("SpoutAuthUpdate", func(t *testing.T) {
```

Comment:
I'm not sure I understand what `SpoutAuthEnabledAfter` does differently than `SpoutAuthUpdate` in terms of exercising the codebase?